examples/wakaama/README.md

```diff
@@ -0,0 +1,90 @@
+# Wakaama LWM2M example client
+
+This application starts a
+[LWM2M](https://wiki.openmobilealliance.org/display/TOOL/What+is+LwM2M) client
+on the node with instances of the following objects:
+- [Security object](http://www.openmobilealliance.org/tech/profiles/LWM2M_Security-v1_0.xml)
+- [Server object](http://www.openmobilealliance.org/tech/profiles/LWM2M_Server-v1_0.xml)
+- [Access control object](http://www.openmobilealliance.org/tech/profiles/LWM2M_Access_Control-v1_0_2.xml)
+- [Device object](http://www.openmobilealliance.org/tech/profiles/LWM2M_Device-v1_0_3.xml)
+- [Light control object](http://www.openmobilealliance.org/tech/profiles/lwm2m/3311.xml)
+
+It will create an instance of the Light control object per LED (LED0 and LED1)
+if available. The state of the LED can be changed from the LWM2M server or by
+using a shell command in which case the state of it will be reflected on the
+LWM2M server.
+
+The application is based on the Eclipse Wakaama
+[example client](https://github.com/eclipse/wakaama/tree/master/examples/client)
+.
+
+## Usage
+
+### Setting up a LWM2M Test Server
+To test the client a LWM2M server where to register is needed.
+[Eclipse Leshan](https://github.com/eclipse/leshan) demo is a good option for
+running one locally.
+
+To run the demo server:
+```shell
+wget https://hudson.eclipse.org/leshan/job/leshan/lastSuccessfulBuild/artifact/leshan-server-demo.jar
+
+java -jar ./leshan-server-demo.jar
+```
+It will output the addresses where it is listening:
+```
+INFO LeshanServer - LWM2M server started at coap://0.0.0.0/0.0.0.0:5683 coaps://0.0.0.0/0.0.0.0:5684
+INFO LeshanServerDemo - Web server started at http://0.0.0.0:8080/.
+```
+
+#### Bootstrap server
+LWM2M provides a bootstrapping mechanism to provide the clients with information
+to register to one or more servers. To test this mechanism both the previous server and a bootstrap server should be running. Eclipse Leshan also provides a bootstrap server demo.
+
```

Comment:
Might be worthwhile to mention here that the bootstrap server is off by default. I had to check the value of LWM2M_SERVER_IS_BOOTSTRAP in lwm2m.h to be sure.

Reply:
Sure, I will add that